lib/FunctionIsolation/IsolateFunctions.cpp

```diff
@@ -667,7 +674,9 @@ bool IFI::handleDirectBoundary(const Boundary &TheBoundary,
     }
   }
 
-  revng_assert(Consumed == TheBoundary.Successors.Addresses.size());
+  // TODO: we may need to review how extra successors
+  // (not present within the model) are handled.
+  revng_assert(Consumed <= TheBoundary.Successors.Addresses.size());
```

Comment:
MEMOTOSELF: revert this change and investigate the situation.

Reply:
Could `FunctionIsolation: handle extra successors` work as commit message?

Reply:
Please don't mark resolved notes for myself.
I need to check this before merging.